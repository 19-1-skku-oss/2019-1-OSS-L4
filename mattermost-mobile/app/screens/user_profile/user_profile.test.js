// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';
import {shallow} from 'enzyme';

import Preferences from 'mattermost-redux/constants/preferences';

import UserProfile from './user_profile.js';
import BotTag from 'app/components/bot_tag';

jest.mock('react-intl');
jest.mock('app/utils/theme', () => {
    const original = require.requireActual('app/utils/theme');
    return {
        ...original,
        changeOpacity: jest.fn(),
    };
});

describe('user_profile', () => {
    const actions = {
        setChannelDisplayName: jest.fn(),
        makeDirectChannel: jest.fn(),
        loadBot: jest.fn(),
    };
    const baseProps = {
        actions,
        config: {
            ShowEmailAddress: true,
        },
        teammateNameDisplay: 'username',
        navigator: {
            resetTo: jest.fn(),
        },
        teams: [],
        theme: Preferences.THEMES.default,
        enableTimezone: false,
        militaryTime: false,
    };

    const user = {
        email: 'test@test.com',
        first_name: 'test',
        id: '4hzdnk6mg7gepe7yze6m3domnc',
        last_name: 'fake',
        nickname: 'nick',
        username: 'fred',
        is_bot: false,
    };

    test('should match snapshot', async () => {
        const wrapper = shallow(
            <UserProfile
                {...baseProps}
                user={user}
            />,
            {context: {intl: {formatMessage: jest.fn()}}},
        );
        expect(wrapper.getElement()).toMatchSnapshot();
    });

    test('should contain bot tag', async () => {
        const botUser = {
            email: 'test@test.com',
            first_name: 'test',
            id: '4hzdnk6mg7gepe7yze6m3domnc',
            last_name: 'fake',
            nickname: 'nick',
            username: 'fred',
            is_bot: true,
        };

        const wrapper = shallow(
            <UserProfile
                {...baseProps}
                user={botUser}
            />,
            {context: {intl: {formatMessage: jest.fn()}}},
        );
        expect(wrapper.containsMatchingElement(
            <BotTag
                show={true}
                theme={baseProps.theme}
            />
        )).toEqual(true);
    });
});
