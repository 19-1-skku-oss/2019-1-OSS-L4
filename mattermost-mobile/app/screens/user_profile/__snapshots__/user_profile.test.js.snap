// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user_profile should match snapshot 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Connect(StatusBar) />
  <ScrollViewMock
    style={
      Object {
        "backgroundColor": "#ffffff",
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "padding": 25,
        }
      }
    >
      <Connect(ProfilePicture)
        size={150}
        statusBorderWidth={6}
        statusSize={40}
        userId="4hzdnk6mg7gepe7yze6m3domnc"
      />
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginTop": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#3d3c40",
              "fontSize": 17,
              "fontWeight": "600",
            }
          }
        >
          fred
        </Text>
        <BotTag
          show={false}
          theme={
            Object {
              "awayIndicator": "#ffbc42",
              "buttonBg": "#166de0",
              "buttonColor": "#ffffff",
              "centerChannelBg": "#ffffff",
              "centerChannelColor": "#3d3c40",
              "codeTheme": "github",
              "dndIndicator": "#f74343",
              "errorTextColor": "#fd5960",
              "linkColor": "#2389d7",
              "mentionBj": "#ffffff",
              "mentionColor": "#145dbf",
              "mentionHighlightBg": "#ffe577",
              "mentionHighlightLink": "#166de0",
              "newMessageSeparator": "#ff8800",
              "onlineIndicator": "#06d6a0",
              "sidebarBg": "#145dbf",
              "sidebarHeaderBg": "#1153ab",
              "sidebarHeaderTextColor": "#ffffff",
              "sidebarText": "#ffffff",
              "sidebarTextActiveBorder": "#579eff",
              "sidebarTextActiveColor": "#ffffff",
              "sidebarTextHoverBg": "#4578bf",
              "sidebarUnreadText": "#ffffff",
              "type": "Mattermost",
            }
          }
        />
      </View>
      <Text
        style={
          Object {
            "color": "#3d3c40",
            "fontSize": 15,
            "marginTop": 15,
          }
        }
      >
        @fred
      </Text>
    </View>
    <View
      style={
        Object {
          "marginBottom": 25,
          "marginHorizontal": 15,
        }
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": undefined,
              "fontSize": 13,
              "fontWeight": "600",
              "marginBottom": 10,
              "marginTop": 25,
            }
          }
        >
          USERNAME
        </Text>
        <Text
          style={
            Object {
              "color": "#3d3c40",
              "fontSize": 15,
            }
          }
        >
          fred
        </Text>
      </View>
      <View>
        <Text
          style={
            Object {
              "color": undefined,
              "fontSize": 13,
              "fontWeight": "600",
              "marginBottom": 10,
              "marginTop": 25,
            }
          }
        >
          NICKNAME
        </Text>
        <Text
          style={
            Object {
              "color": "#3d3c40",
              "fontSize": 15,
            }
          }
        >
          nick
        </Text>
      </View>
    </View>
    <userProfileRow
      action={[Function]}
      defaultMessage="Send Message"
      icon="paper-plane-o"
      iconColor="rgba(0, 0, 0, 0.7)"
      iconSize={15}
      iconType="fontawesome"
      textColor="#000"
      textId="mobile.routes.user_profile.send_message"
      theme={
        Object {
          "awayIndicator": "#ffbc42",
          "buttonBg": "#166de0",
          "buttonColor": "#ffffff",
          "centerChannelBg": "#ffffff",
          "centerChannelColor": "#3d3c40",
          "codeTheme": "github",
          "dndIndicator": "#f74343",
          "errorTextColor": "#fd5960",
          "linkColor": "#2389d7",
          "mentionBj": "#ffffff",
          "mentionColor": "#145dbf",
          "mentionHighlightBg": "#ffe577",
          "mentionHighlightLink": "#166de0",
          "newMessageSeparator": "#ff8800",
          "onlineIndicator": "#06d6a0",
          "sidebarBg": "#145dbf",
          "sidebarHeaderBg": "#1153ab",
          "sidebarHeaderTextColor": "#ffffff",
          "sidebarText": "#ffffff",
          "sidebarTextActiveBorder": "#579eff",
          "sidebarTextActiveColor": "#ffffff",
          "sidebarTextHoverBg": "#4578bf",
          "sidebarUnreadText": "#ffffff",
          "type": "Mattermost",
        }
      }
      togglable={false}
    />
  </ScrollViewMock>
</View>
`;
